app/frontend/wallet/shelley/shelley-bitbox02-crypto-provider.ts

```diff
@@ -0,0 +1,350 @@
+import {
+  CardanoInput,
+  CardanoOutput,
+  CardanoCertificate,
+  CardanoShelleyWitness,
+  BitBox02API,
+} from 'bitbox02-api'
+import * as cbor from 'borc'
+import CachedDeriveXpubFactory from '../helpers/CachedDeriveXpubFactory'
+import {ShelleySignedTransactionStructured, cborizeTxWitnesses} from './shelley-transaction'
+import {hasRequiredVersion} from './helpers/version-check'
+import {BITBOX02_ERRORS, BITBOX02_VERSIONS} from '../constants'
+
+import debugLog from '../../helpers/debugLog'
+import derivationSchemes from '../helpers/derivation-schemes'
+
+import {
+  CryptoProvider,
+  CryptoProviderFeature,
+  BIP32Path,
+  HexString,
+  DerivationScheme,
+  AddressToPathMapper,
+  CertificateType,
+} from '../../types'
+import {
+  CryptoProviderType,
+  Network,
+  NetworkId,
+  TxCertificate,
+  TxInput,
+  TxOutput,
+  TxShelleyWitness,
+} from '../types'
+import {TxSigned, TxAux, CborizedCliWitness} from './types'
+import {
+  InternalError,
+  InternalErrorReason,
+  UnexpectedError,
+  UnexpectedErrorReason,
+} from '../../errors'
+
+let _activeBitBox02: BitBox02API | null = null
+
+type CryptoProviderParams = {
+  network: Network
+  config: any
+}
+
+const ShelleyBitBox02CryptoProvider = async ({
+  network,
+  config,
+}: CryptoProviderParams): Promise<CryptoProvider> => {
+  const bitbox02API = await import(/* webpackChunkName: "bitbox02-api" */ 'bitbox02-api')
```

Comment:
Can you please add an explanatory comment of what this does and why it's needed?